File: 프로그래머스/2/42842. 카펫/카펫.py
def solution(brown, yellow):
    total = brown + yellow
    for y in range(1, int(yellow ** 0.5) + 1):
        if yellow % y == 0:
            x = yellow // y
            
            if (x + 2) * (y + 2) == total:
                return [x + 2, y + 2]

File: 프로그래머스/lv2/42842. 카펫/카펫.cpp
#include <string>
#include <vector>
#include <cmath>

using namespace std;

vector<int> solution(int brown, int yellow) {
    int total = brown + yellow;
    for (int i = 1; i <= sqrt(yellow); i++) {
        if (yellow % i == 0) {
            int x = yellow / i;
            
            if ((x + 2) * (i + 2) == total) {
                return {x + 2, i + 2};
            }
        }
    }
}